### PR TITLE
Review/Rework of English Language Readability

### PR DESCRIPTION
**Pull request by mancoast**
Greetings!

This Pull request exists for peer review of my proposed edits from this first contribution to the core guidelines.

Looking forward to further review and editing.

Thanks,
coast

### CODE REVIEW

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -1831,7 +1831,7 @@ Alternatively, we could use concepts (as defined by the ISO TS) to define the no
 
 ##### Note
 
-How many arguments are too many? Four arguments is a lot.
+How many arguments are too many? Four arguments are a lot.
```
Comment (jwakely) [inline CppCoreGuidelines.md]: Grammatically this is correct, but I find myself disliking it.

File: CppCoreGuidelines.md [hunk 2]
```diff
@@ -618,7 +618,7 @@ Leaving hard-to-detect errors in a program is asking for crashes and bad results
 
 ##### Note
 
-Ideally we catch all errors (that are not errors in the programmer's logic) at either compile-time or run-time. It is impossible to catch all errors at compile time and often not affordable to catch all remaining errors at run time. However, we should endeavor to write programs that in principle can be checked, given sufficient resources (analysis programs, run-time checks, machine resources, time).
+Ideally, we catch all errors (that are not errors in the programmer's logic) at either compile-time or run-time. It is impossible to catch all errors at compile time and often not affordable to catch all remaining errors at run time. However, we should endeavor to write programs that in principle can be checked, given sufficient resources (analysis programs, run-time checks, machine resources, time).
```
Comment (galik) [inline CppCoreGuidelines.md]: Line 211 sounds wrong in the new version. "It is typically infeasible to convert the entire code base at once." sounds better to me.

File: CppCoreGuidelines.md [hunk 3]
```diff
@@ -1560,7 +1560,7 @@ What is an error?
 An error means that the function cannot achieve its advertised purpose (including establishing postconditions).
 Calling code that ignores an error could lead to wrong results or undefined systems state.
 For example, not being able to connect to a remote server is not by itself an error:
-the server can refuse a connection for all kinds of reasons, so the natural thing is to return a result that the caller always has to check.
+the server can refuse a connection for all kinds of reasons, so the natural thing is to return a result that the caller always should check.
```
Comment (galik) [inline CppCoreGuidelines.md]: "should always" rather than "always should"

File: CppCoreGuidelines.md [hunk 4]
```diff
@@ -1781,7 +1781,7 @@ Complex initialization can lead to undefined order of execution.
 
 Since `x` and `y` are in different translation units the order of calls to `f()` and `g()` is undefined;
 one will access an uninitialized `const`.
-This particular example shows that the order-of-initialization problem for global (namespace scope) objects is not limited to global *variables*.
+This exact example shows that the order-of-initialization problem for global (namespace scope) objects is not limited to global *variables*.
```
Comment (galik) [inline CppCoreGuidelines.md]: There is only one example so "this example shows..." or simply "this shows..."

File: CppCoreGuidelines.md (hunk 2 above)
Comment (jwakely) [inline CppCoreGuidelines.md]: That loses the "large" part. It's entirely feasible to convert a small code base all at once. Maybe "to completely convert a large code base all at once."

**Comment by mancoast** [comment]
@jwakely @galik Edits have been updated to reflect your Suggestions.

Thanks!

**Comment by AndrewPardoe** [comment]
Thank you!

**Pull request merged**